tezos/context/src/kv_store/readonly_ipc.rs

```diff
@@ -136,6 +137,94 @@ impl KeyValueStoreBackend for ReadonlyIpcBackend {
     fn synchronize_strings_into(&self, _string_interner: &mut StringInterner) {
         // Readonly protocol runner doesn't update strings.
     }
+
+    fn get_object(
+        &self,
+        object_ref: ObjectReference,
+        storage: &mut Storage,
+        strings: &mut StringInterner,
+    ) -> Result<Object, DBError> {
+        self.get_object_bytes(object_ref, &mut storage.data)?;
+
+        let bytes = std::mem::take(&mut storage.data);
+
+        let object_header: ObjectHeader = ObjectHeader::from_bytes([bytes[0]; 1]);
+
+        let result = if object_header.get_persistent() {
+            persistent::deserialize_object(&bytes, object_ref.offset(), storage, strings, self)
+        } else {
+            in_memory::deserialize_object(&bytes, storage, strings, self)
+        };
+
+        storage.data = bytes;
+
+        result.map_err(Into::into)
+    }
+
+    fn get_object_bytes<'a>(
+        &self,
+        object_ref: ObjectReference,
+        buffer: &'a mut Vec<u8>,
+    ) -> Result<&'a [u8], DBError> {
+        if let Some(hash_id) = object_ref
+            .hash_id_opt()
+            .and_then(|h| h.get_readonly_id().ok()?)
+        {
+            let bytes = self
+                .hashes
+                .get_value(hash_id)?
+                .ok_or(DBError::HashNotFound { object_ref })?;
+            buffer.clear();
+            buffer.extend_from_slice(bytes);
+            return Ok(buffer);
+        };
+
+        self.client
+            .get_object_bytes(object_ref, buffer)
+            .map_err(|reason| DBError::IpcAccessError { reason })
+    }
+
+    fn commit(
```

Comment:
I might be wrong but I thought `commit`, on the readonly-ipc backend, was used to produce the hash ?
This `ReadonlyIpcBackend::commit` doesn't write anything to storage.
See `WorkingTree::prepare_commit`, it takes a parameter `serialize_function` which is `None` in that case

Reply:
@sebastiencs I don't think this will ever be called (readonly protocol runners will not call commit). In any case, it doesn't hurt for it to work like this, but please add a comment clarifying that readonly-ipc never really uses commit to make that clear.